v2/clients/http/deviceprofile_test.go

```diff
@@ -90,13 +94,31 @@ func TestAddDeviceProfileByYaml(t *testing.T) {
 		_, _ = w.Write(res)
 	}))
 	defer ts.Close()
-
 	client := NewDeviceProfileClient(ts.URL)
 	_, b, _, _ := runtime.Caller(0)
-	res, err := client.AddByYaml(context.Background(), filepath.Dir(b)+"/data/sample-profile.yaml")
-	require.NoError(t, err)
-	require.NotNil(t, res)
-	require.Equal(t, requestId, requestId)
+
+	tests := []struct {
+		name          string
+		filePath      string
+		errorExpected bool
+	}{
+		{name: "Add device profile by yaml file", filePath: filepath.Dir(b) + "/data/sample-profile.yaml", errorExpected: false},
+		{name: "file not found error", filePath: filepath.Dir(b) + "/data/file-not-found.yaml", errorExpected: true},
+	}
+	for _, testCase := range tests {
+		t.Run(testCase.name, func(t *testing.T) {
+			res, err := client.AddByYaml(context.Background(), testCase.filePath)
+			if testCase.errorExpected {
+				assert.True(t, errors.Is(err, os.ErrNotExist))
```

Comment:
Use `require.True()` so test fails immediately. i.e. following assert are invalid to do if not an error. Same in other test.

Reply:
Modified.